File: 2019/6/lib/aoc/parser.ex
defmodule AOC.Parser do
  def parse do
    {:ok, binary} = File.read("./input.txt")
    _parse(binary)
  end

  def parse(binary) do
    _parse(binary)
  end

  defp _parse(binary) do
    binary
    |> String.trim
    |> String.split("\n")
    |> Enum.reduce(%{}, &formatted/2)
  end

  def formatted(line, acc) do
    [parent, child] = line |> String.trim |> String.split(")")
    Map.update(acc, parent, [child], &([child | &1]))
  end
end

File: 2020/8/lib/parser.rb
# frozen_string_literal: true

class Parser
  def self.parse(input = nil)
    input ||= File.read(File.join(File.dirname(__FILE__), "../input.txt"))

    input
      .strip
      .split("\n")
      .each_with_object([]) do |l, acc|
        inst, value = l.split
        acc << {
          instruction: inst,
          value: value.to_i
        }
      end
  end
end
